Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h

```diff
@@ -116,8 +116,10 @@ class RAPYUTASIMULATIONPLUGINS_API URRRobotROS2Interface : public UObject
     /**
      * @brief Setup ROS Params, overridable by child classes to config custom ROS2 Interface's params
      */
-    UFUNCTION()
-    virtual void SetupROSParams();
+    UFUNCTION(BlueprintNativeEvent, BlueprintCallable)
+    void SetupROSParams();
+
+    virtual void SetupROSParams_Implementation();

```

Comment:
```suggestion
    UFUNCTION()
    virtual void SetupROSParams();

    UFUNCTION(BlueprintNativeEvent, BlueprintCallable)
    void BPSetupROSParams_Implementation();
```
So we could keep various child cpp classes's API header without having to update them with `_Implementation`, which is BP coupling.

Reply:
it is used only by SBROS2Interface.

Reply:
Yeah it is used by Skeletal robot also, but I get the idea to make this supported by child BP.
Since this is one of the core class, ideally I'd still prefer an agnostic style of API, so either child cpp or BP class could use, but it might not be fully supported by UE.